[WIP]Add Migrate functionality in the worker generic actuator

vlvasilev opened the pull request:
**What this PR does / why we need it**:
With this PR when we annotate a Worker with "gardener.cloud/operation=migrate" it will delete the machine-controll-manager and will make shallow delete(it will remove the finalizers) all of the machine related resource. The Virtual Machines (Shoot's Nodes) will be untouched.
**Which issue(s) this PR fixes**:
Part of [#1631](https://github.com/gardener/gardener/issues/1631)

**Special notes for your reviewer**:

**Release note**:
<!--  Write your release note:
1. Enter your release note in the below block.
2. If no release note is required, just write "NONE" within the block.

Format of block header: <category> <target_group>
Possible values:
- category:       improvement|noteworthy|action
- target_group:   user|operator|developer
-->
```improvement operator
Implement handling of "gardener.cloud/operation=migrate" annotation
```

rfranzke commented:
We have moved the extension controllers into dedicated repositories.

Please open a new PR if you have changes in folder different than `controllers` in this repository, and another PR with only your changes in the respective provider specific repositories:
* https://github.com/gardener/gardener-extension-provider-alicloud
* https://github.com/gardener/gardener-extension-provider-aws
* https://github.com/gardener/gardener-extension-provider-azure
* https://github.com/gardener/gardener-extension-provider-gcp
* https://github.com/gardener/gardener-extension-provider-openstack
* https://github.com/gardener/gardener-extension-provider-vsphere

Thank you and sorry for the inconveniences.
/close

rfranzke commented:
Reopening as you only have changes in the `pkg` folder. Sorry, didn't see this :)
/reopen

vlvasilev commented:
@rfranzke I have a problem testing the controllers.
I have set replacing statement in the github.com/gardener/gardener-extension-provider-gcp/go.mod 
replace (
    github.com/gardener/gardener-extensions => ../../gardener/gardener-extensions
    ...)
I am using golang 1.13.4 and after executing "go mod vendor" I get the following error:

github.com/gardener/gardener-extension-provider-gcp/hack imports
        github.com/gardener/gardener-extensions/.ci: **malformed module path** "github.com/gardener/gardener-extensions/.ci": **leading dot in path element**
github.com/gardener/gardener-extension-provider-gcp/hack imports
        github.com/gardener/gardener-extensions/hack/api-reference/template: **module github.com/gardener/gardener-extensions@latest found (v1.3.0, replaced by ../../gardener/gardener-extensions), but does not contain package github.com/gardener/gardener-extensions/hack/api-reference/template**
github.com/gardener/gardener-extension-provider-gcp/hack imports
        github.com/gardener/gardener-extensions/hack/code-generator: module github.com/gardener/gardener-extensions@latest found (v1.3.0, replaced by ../../gardener/gardener-extensions), but does not contain package github.com/gardener/gardener-extensions/hack/code-generator
github.com/gardener/gardener-extension-provider-gcp/cmd/gardener-extension-provider-gcp/app imports
        github.com/gardener/gardener-extensions/pkg/controller/worker imports
        github.com/gardener/gardener/pkg/apis/extensions/v1beta1: module github.com/gardener/gardener@latest found (v0.35.2), but does not contain package github.com/gardener/gardener/pkg/apis/extensions/v1beta1


If I try to run it inside a container with golang 1.12.0 (`GO111MODULE=on go mod vendor`) and it starts well at the beginning but after a time it return a error:
**go: finding github.com/spf13/pflag v1.0.5
go: error loading module requirements**

If I use another golang binary version 1.12.1 it gives mi the following error:
**go: modules disabled inside GOPATH/src by GO111MODULE=auto; see 'go help modules'**

So how to put my changes in `gardener-extensions` and to have it in `gardener-extension-provider-gcp` ?

rfranzke commented:
> github.com/gardener/gardener-extensions/.ci: malformed module path "github.com/gardener/gardener-extensions/.ci": leading dot in path element

This indicates that you have not rebased. Please rebase on `master` and try again.
